app/javascript/packs/readingList.jsx

```diff
@@ -1,11 +1,21 @@
 import { h, render } from 'preact';
+import { getUserDataAndCsrfToken } from '../chat/util';
 import { ReadingList } from '../readingList/readingList';
 
 function loadElement() {
-  const root = document.getElementById('reading-list');
-  if (root) {
-    render(<ReadingList />, root, root.firstElementChild);
-  }
+  getUserDataAndCsrfToken().then(({ currentUser }) => {
```

Comment:
As in the history component, we can initialize it after the user data and the token have been retrieved, instead of calling a function on an interval inside the component